source/common/stats/thread_local_store.cc

```diff
@@ -193,9 +184,17 @@ StatType& ThreadLocalStoreImpl::ScopeImpl::safeMakeStat(
     const std::string& name, StatMap<std::shared_ptr<StatType>>& central_cache_map,
     MakeStatFn<StatType> make_stat, StatMap<std::shared_ptr<StatType>>* tls_cache) {
 
+  const char* stat_key = name.c_str();
+  std::unique_ptr<std::string> truncation_buffer;
+  absl::string_view truncated_name = parent_.truncateStatNameIfNeeded(name);
+  if (truncated_name.size() < name.size()) {
+    truncation_buffer = std::make_unique<std::string>(std::string(truncated_name));
+    stat_key = truncation_buffer->c_str(); // must be nul-terminated.
```

Comment:
would be nice to say the why, just in case this code needs to change and we need to preserve certain assurances.